src/main/java/uk/gov/pay/directdebit/payments/services/PaymentService.java

```diff
@@ -113,36 +105,21 @@ Payment submitPaymentToProvider(Payment payment) {
         return paymentSubmittedToProviderFor(submittedPayment);
     }
 
-    public PaymentResponse createPaymentResponseWithAllLinks(Payment payment, String accountExternalId, UriInfo uriInfo) {
-        List<Map<String, Object>> dataLinks = new ArrayList<>();
-
-        dataLinks.add(createLink("self", GET, selfUriFor(uriInfo, CHARGE_API_PATH, accountExternalId, payment.getExternalId())));
-
-        if (!payment.getState().toExternal().isFinished()) {
-            Token token = tokenService.generateNewTokenFor(payment.getMandate());
-            dataLinks.add(createLink("next_url",
-                    GET,
-                    nextUrl(directDebitConfig.getLinks().getFrontendUrl(), "secure", token.getToken())));
-            dataLinks.add(createLink("next_url_post",
-                    POST,
-                    nextUrl(directDebitConfig.getLinks().getFrontendUrl(), "secure"),
-                    APPLICATION_FORM_URLENCODED,
-                    ImmutableMap.of("chargeTokenId", token.getToken())));
-        }
-        return PaymentResponse.from(payment, dataLinks);
+    public CollectPaymentResponse createPaymentResponseWithAllLinks(Payment payment) {
```

Comment:
There's no point to this method anymore as links aren't set on the payment. This method can be deleted.